src/session/sessionAccountMiddleware.ts

```
@@ -9,22 +9,14 @@ export const toSessionAccountMiddleware = (server: ApiServer): Middleware => {
 		}
 
 		console.log('[sessionAccountMiddleware]', req.session.name); // TODO logging
```

Because the purpose of this middleware is to either add the `req.accountSession` or clear the `req.session`, I think we should do a check for `req.accountSession` here - if it exists, let's throw an error for ourselves.

---

So I think this is the last blocking piece, but I'm not really sure I understand what you're asking for/describing here. Could you draft a more concrete example of what you have in mind?

---

This is what I was thinking, does it make sense?

```ts
import send from '@polka/send-type';

import type {ApiServer, Middleware} from '../server/ApiServer.js';

export const toSessionAccountMiddleware = (server: ApiServer): Middleware => {
	return async (req, res, next) => {
		if (req.accountSession) {
			return send(res, 500, {reason: 'invalid server configuration'});
		}
```

---

Incidentally, I think this change will make CI pass.